### PR TITLE
smarter script for creating Android project

### PR DESCRIPTION
**Pull request by longlene**
1、You can add the create-android-project.sh  to the PATH, so you can run it to create android project in the current directory if you like. Inaddition, you can put  the script anywhere you like and run it by using ./create-android-project.sh.
2、After launching the script, you can just press enter to create a default project and then call the build_native.sh to compile it.
3、Environment variables: NDK_ROOT、ANDROID_SDK_ROOT、COCOS2DX_ROOT 
are needed, or you can set them in the script.

For example:

2013-06-04-224030_516x299_scrot

That's all.

### CODE REVIEW

**Comment by minggo** [comment]
Looks great, thank you @longlene

File: template/android/copy_files.sh [hunk 1]
```diff
@@ -1,60 +1,56 @@
...
 # from HelloWorld copy src and jni to APP_DIR
-copy_src_and_jni(){
-    cp -rf $HELLOWORLD_ROOT/proj.android/jni $APP_DIR/proj.android
-    cp -rf $HELLOWORLD_ROOT/proj.android/src $APP_DIR/proj.android
+copy_src_and_jni() {
+    cp -rf $HELLOWORLD_ROOT/proj.android/{jni,src} $APP_DIR/proj.android
+
+    # cocos2d-x java lib
+    mkdir -p $APP_DIR/proj.android/src/org/cocos2dx/lib/
+    cp -f $COCOS2DX_ROOT/cocos2dx/platform/android/java/src/org/cocos2dx/lib/*.java $APP_DIR/proj.android/src/org/cocos2dx/lib/
```
Comment (minggo) [inline template/android/copy_files.sh]: Should not copy java files.
Java files is a separate lib, you should reference it.

Comment (longlene) [inline template/android/copy_files.sh]: Thanks! Got it! I've remove those lines.

**Comment by minggo** [comment]
Thanks @longlene.
Could you please merge conflict of this pull request?

File: tools/bindings-generator [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 716d9156b845173f14e38fae10a76c54a11b64ef
+Subproject commit 55541ddbc4be0d4a6594a59772d36526eedbf373
```
Comment (dumganhar) [inline tools/bindings-generator]: Why to change the submodule reference of `bindings-generator`?

File: external/emscripten [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit 0ad87244178badf26cd5c8e0ed88116e87026472
+Subproject commit 218cfaa4236a9f5ffa4abb4628e612299f5fc171
```
Comment (dumganhar) [inline external/emscripten]: ??

File: samples/Javascript/Shared [hunk 4]
```diff
@@ -1 +1 @@
-Subproject commit 38374930ae707c70611ad1fcb3a5af91d50d72d0
+Subproject commit 91d853f5c72df3137e2f3eb68229b426a3252e5d
```
Comment (dumganhar) [inline samples/Javascript/Shared]: ??

File: scripting/javascript/bindings/generated [hunk 5]
```diff
@@ -1 +1 @@
-Subproject commit 87d3aad2ca84f88ee12d47f28db78d3f98349ddc
+Subproject commit 2b5f3a92aca4754f98a97412220427647d82c921
```
Comment (dumganhar) [inline scripting/javascript/bindings/generated]: ??

File: samples/Javascript/Shared (hunk 4 above)
Comment (dumganhar) [inline samples/Javascript/Shared]: After you pull the latest codes, please make sure you had done

``` sh
git submodule update
```

Comment (longlene) [inline samples/Javascript/Shared]: Sorry for deletion of last msg.
I've updated the submodule and merged the conflict file.
Thanks a lot!

**Comment by minggo** [comment]
Thanks.

**Pull request merged**